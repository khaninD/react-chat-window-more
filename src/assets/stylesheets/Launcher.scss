.rc-laucher {
  width: 60px;
  height: 60px;
  background-color: #4e8cff;
  background-position: center;
  background-repeat: no-repeat;
  position: fixed;
  right: 25px;
  bottom: 25px;
  border-radius: 50%;
  box-shadow: none;
  transition: box-shadow 0.2s ease-in-out;
  &:hover {
    box-shadow: 0 0px 27px 10px rgba(0,0,0,0.22);
  }
  .rc-laucher__logo {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    .rc-laucher__logo-base {
      opacity: 1;
      width: 100%;
      transition: all 0.2s ease-in-out;
      cursor: pointer
    }
    .rc-laucher__logo-close {
      position: absolute;
      opacity: 0;
      width: 20px;
      height: 20px;
      cursor: pointer;
      transition: opacity 0.2s ease-in-out;
    }
  }
  .rc-laucher__chat-container {
    position: fixed;
    right: 25px;
    bottom: 0;
    opacity: 0;
    transition: all 0.2s ease-in-out;
  }
}

.rc-laucher.opened {
  .rc-laucher__logo-base {
    opacity: 0;
    transform: rotate(-90deg);
  }
  .rc-laucher__logo-close {
    opacity: 1;
  }

  .rc-laucher__chat-container {
    opacity: 1;
    bottom: 100px;
  }
}