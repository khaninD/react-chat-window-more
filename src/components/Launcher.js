import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import isFunction from 'lodash/isFunction'
import launcherIcon from './../assets/images/logo-no-bg.svg'
import closeIcon from './../assets/images/close-icon.png'
import { LaucherConstants } from '../constants'
import Window from './Window'

const {
  closeButtonAlt,
  logoAlt
} = LaucherConstants
export default class Launcher extends Component {
  static defaultProps = {
    customStyle: null,
    customChatIcon: [
      <img key={0} src={launcherIcon} className='rc-laucher__logo-base' alt={logoAlt} />,
      <img key={1} src={closeIcon} className='rc-laucher__logo-close' alt={closeButtonAlt} />
    ],
    customClassNames: null
  }
  static propTypes = {
    customStyle: PropTypes.object,
    customChatIcon: PropTypes.oneOfType([
      PropTypes.element,
      PropTypes.arrayOf(PropTypes.element)
    ]),
    customClassNames: PropTypes.string
  }
  constructor() {
    super()
    this.state = {
      isOpen: false
    }
  }

  handleClick = () => this.setState({ isOpen: !this.state.isOpen })

  render() {
    const { customStyle, customChatIcon, customClassNames } = this.props
    const isOpen = this.props.hasOwnProperty('isOpen') ? this.props.isOpen : this.state.isOpen
    const classList = classNames({
      'rc-laucher': true,
      opened: isOpen
    })
    return (
      <div style={customStyle} className={customClassNames || classList} onClick={this.handleClick}>
        <div className='rc-laucher__logo'>
          {customChatIcon}
        </div>
        <div className='rc-laucher__chat-container'>
          {
            isOpen && <Window />
          }
        </div>
      </div>
    )
  }
}
