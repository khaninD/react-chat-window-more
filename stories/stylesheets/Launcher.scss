.reach-chat-window-more {
  width: 60px;
  height: 60px;
  background-color: #4e8cff;
  background-position: center;
  background-repeat: no-repeat;
  position: fixed;
  right: 25px;
  bottom: 25px;
  border-radius: 50%;
  box-shadow: none;
  transition: box-shadow 0.2s ease-in-out;
  cursor: pointer;
}