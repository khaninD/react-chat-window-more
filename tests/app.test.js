import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import renderer from 'react-test-renderer'
import Launcher from '../src/components/Launcher'
import Icon from '../examples/images/chat.png'

Enzyme.configure({ adapter: new Adapter() })

describe('Test custom styles and classes', () => {
  it('it\'s render App ', () => {
    const wrapper = shallow(<Launcher />)
    expect(wrapper.find('.rc-laucher')).toHaveLength(1)
  })
  it('It\'s render App with custom styles', () => {
    const style = {
      left: 0,
      bottom: 0
    }
    const wrapper = shallow(<Launcher customStyle={style} />)
    expect(wrapper.find('.rc-laucher').prop('style')).toBe(style)
  })
  it('It\'s render App with custom Icon', () => {
    const customIcon = <div className='custom-chat-icon'>{Icon}</div>
    const wrapper = shallow(<Launcher customChatIcon={customIcon} />)
    expect(wrapper.find('.custom-chat-icon')).toHaveLength(1)
  })
  it('It\'s render App with custom classNames', () => {
    const customClassName = 'myCustomClassName'
    const wrapper = shallow(<Launcher customClassNames={customClassName} />)
    expect(wrapper.find(`.${customClassName}`)).toHaveLength(1)
  })
})

describe('Test events', () => {
  it('click on chatIcon', () => {
    const wrapper = shallow(<Launcher />)
    wrapper.find('.rc-laucher').simulate('click')
    expect(wrapper.find('.rc-laucher').hasClass('opened')).toBe(true)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Launcher />)
    wrapper.find('.rc-laucher').simulate('click')
    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })
})