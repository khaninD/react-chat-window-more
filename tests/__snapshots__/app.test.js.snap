// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test events matches snapshot 1`] = `
<div
  className="rc-laucher opened"
  onClick={[Function]}
  style={null}
>
  <div
    className="rc-laucher__logo"
  >
    <img
      alt="react chat logo"
      className="rc-laucher__logo-base"
      src={Object {}}
    />
    <img
      alt="react chat close button"
      className="rc-laucher__logo-close"
      src={Object {}}
    />
  </div>
  <div
    className="rc-laucher__chat-container"
  >
    <div
      className="rc-window"
      style={null}
    />
  </div>
</div>
`;
